### PR TITLE
Enable SortMergeJoin LeftAnti filtered fuzz tests

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/10872.

## Rationale for this change
Fixed the fuzz test, previously the filter was not set up correctly

### CODE REVIEW

File: datafusion/core/tests/fuzz_cases/join_fuzz.rs [hunk 1]
```diff
@@ -217,6 +225,7 @@ async fn test_semi_join_1k() {
 
 #[tokio::test]
 async fn test_semi_join_1k_filtered() {
+    // NLJ vs HJ gives wrong result
```
Comment (comphead) [inline datafusion/core/tests/fuzz_cases/join_fuzz.rs:228]: The HashJoin and NestedLoop Join gives different results for filtered joins fuzz tests. 
In the same time HashJoin and SortMergeJoin results are correct. 

Need to triage what is going on.

File: datafusion/core/tests/fuzz_cases/join_fuzz.rs [hunk 2]
```diff
@@ -239,17 +248,16 @@ async fn test_anti_join_1k() {
     .await
 }
 
-// Test failed for now. https://github.com/apache/datafusion/issues/10872
-#[ignore]
 #[tokio::test]
 async fn test_anti_join_1k_filtered() {
+    // NLJ vs HJ gives wrong result
```
Comment (andygrove) [inline datafusion/core/tests/fuzz_cases/join_fuzz.rs:253]: Can you file an issue for this and reference it here?

File: datafusion/core/tests/fuzz_cases/join_fuzz.rs (hunk 1 above)
Comment (andygrove) [reply datafusion/core/tests/fuzz_cases/join_fuzz.rs:228]: Could you add a reference to the issue with the comment here

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @comphead

**Review by viirya — APPROVED** [review]

**Comment by comphead** [comment]
Thanks everyone for get this reviewed

**Comment by alamb** [comment]
FYI I have seen this test fail now a few times on unrelated PRs: https://github.com/apache/datafusion/issues/11555

**Pull request merged**